pyseir/inference/infer_rt.py

```diff
@@ -0,0 +1,460 @@
+
+            self.times, self.observed_new_cases, self.observed_new_deaths = \
+                load_data.load_new_case_data_by_state(self.state, self.ref_date)
+
+            self.hospital_times, self.hospitalizations, self.hospitalization_data_type = \
+                load_data.load_hospitalization_data_by_state(self.state_obj.abbr, t0=self.ref_date)
+            self.display_name = self.state
+        else:
+            self.agg_level = AggregationLevel.COUNTY
+            self.geo_metadata = load_data.load_county_metadata().set_index('fips').loc[fips].to_dict()
+            self.state = self.geo_metadata['state']
+            self.state_obj = us.states.lookup(self.state)
+            self.county = self.geo_metadata['county']
+            if self.county:
+                self.display_name = self.county + ', ' + self.state
+            else:
+                self.display_name = self.state
+
+            # TODO Swap for new data source.
+            self.times, self.observed_new_cases, self.observed_new_deaths = \
+                load_data.load_new_case_data_by_fips(self.fips, t0=self.ref_date)
+            self.hospital_times, self.hospitalizations, self.hospitalization_data_type = \
+                load_data.load_hospitalization_data(self.fips, t0=self.ref_date)
+
+        logging.info(f'Running Rt Inference for {self.display_name}')
+
+        self.case_dates = [ref_date + timedelta(days=int(t)) for t in self.times]
+        if self.hospitalization_data_type:
+            self.hospital_dates = [ref_date + timedelta(days=int(t)) for t in self.hospital_times]
+
+        self.default_parameters = ParameterEnsembleGenerator(
+            fips=self.fips,
+            N_samples=500,
+            t_list=np.linspace(0, 365, 366)
+        ).get_average_seir_parameters()
+
+        # Serial period = Incubation + 0.5 * Infections
+        self.serial_period = 1 / self.default_parameters['sigma'] + 0.5 * 1 /   self.default_parameters['delta']
+
+        # If we only receive current hospitalizations, we need to account for
+        # the outflow to reconstruct new admissions.
+        if self.hospitalization_data_type is load_data.HospitalizationDataType.CURRENT_HOSPITALIZATIONS:
+            los_general = self.default_parameters['hospitalization_length_of_stay_general']
+            los_icu = self.default_parameters['hospitalization_length_of_stay_general']
+            hosp_rate_general = self.default_parameters['hospitalization_rate_general']
+            hosp_rate_icu = self.default_parameters['hospitalization_rate_icu']
+            icu_rate = hosp_rate_icu / hosp_rate_general
+            flow_out_of_hosp = self.hospitalizations[:-1] * ((1 - icu_rate) / los_general + icu_rate / los_icu)
+            # We are attempting to reconstruct the cumulative hospitalizations.
+            self.hospitalizations = np.diff(self.hospitalizations) + flow_out_of_hosp
```

Comment:
Should the sign be flipped on `flow_out_of_hospital`?
```suggestion
            self.hospitalizations = np.diff(self.hospitalizations) - flow_out_of_hosp
```

Reply:
No, because we are trying to reconstruct the new cumulative cases, so we have to add back the cases that left first, and then compute the derivative.